src/web/app/pages-admin/admin-search-page/admin-search-page.component.spec.ts

```diff
@@ -351,14 +353,16 @@ describe('AdminSearchPageComponent', () => {
         createdAt: 'Tue, 08 Feb 2022, 08:23 AM +00:00',
         registeredAt: 'Not Registered Yet',
         showLinks: true,
+        hasRegistered: false,
       }, {
         name: 'name2',
         email: 'email2',
         institute: 'institute2',
         registrationLink: 'registrationLink2',
         createdAt: 'Tue, 08 Feb 2022, 08:23 AM +00:00',
-        registeredAt: 'Not Registered Yet',
+        registeredAt: 'Wed, 09 Feb 2022, 10:23 AM +00:00',
         showLinks: true,
+        hasRegistered: true,
```

Comment:
do we need both `hasRegistered` and `registeredAt`? I think `registeredAt` can tell whether the user has registered or not

Reply:
Not strictly necessary. We can check if the string is equals to "Not Registered Yet" and compare that way. 

I did it this way because I figured it will be less bug prone. The assumption is that the string is either a timestamp or the string 'Not Registered Yet'. My concern is that it is possible that someone adds a third state in the future, maybe 'Not Registered Yet (...)'. In this case, the equality check might fail, and it is not a guarantee that the tests might fail as well.

Do you think this is valid, I can still change it if it's unnecessary.

Reply:
I feel like we can keep `registeredAt: Date | null` instead of plain text, so registeredAt = null would mean hasRegistered = false, and we can parse the date for displaying

Reply:
> I feel like we can keep `registeredAt: Date | null` instead of plain text, so registeredAt = null would mean hasRegistered = false, and we can parse the date for displaying

This makes sense, I'll make the changes :)